docs/tutorials/sources/file.rst

```diff
@@ -0,0 +1,216 @@
+Simple source for new file types
+================================
+
+This tutorial will help you with implementing your own ``Source`` for new file
+types. You may want to do this if you have some specific file formats from which
+you want to load and save data from.
+
+Here we will go through implementing a source for ``.ini`` file formats.
+
+Create the Package
+------------------
+
+To create a new source we first create a new Python package. DFFML has a script
+to do it for you.
+
+.. code-block:: console
+
+    $ dffml service dev create source dffml-source-ini
+    $ cd dffml-source-ini
+    $ ls
+    dffml_source_ini  LICENSE  MANIFEST.in  pyproject.toml  README.md  setup_common.py  setup.py  tests
+
+We will start writing our source in ./dffml_source_ini/misc.py
+
+About INI files
+---------------
+
+An INI file is a configuration file used by Operating Systems and programs
+to initialize program settings. It contains sections for settings and preferences
+(delimited by a string in square brackets) with each section containing one
+or more name and value parameters.
+
+Import modules
+--------------
+
+**dffml_source_ini/misc.py**
+
+.. code-block:: python
+
+    from configparser import ConfigParser
+
+    from dffml import config, entrypoint, Record, FileSource, MemorySource, parser_helper
+
+Here we are importing some common modules which will be required. The ``configparser``
+module will be helpful in parsing INI files. The ``FileSource`` and ``MemorySource``
+will be used as base class for our new Source. The config was imported to set the
+configuration options for our new INISource. The entrypoint will be used to add the
+entrypoint to our INISource. A ``Record`` is a unique entry in a source.
+
+Add configuration
+-----------------
+
+**dffml_source_ini/misc.py**
+
+.. literalinclude:: ../../../dffml/source/ini.py
+    :lines: 11-15
+
+Here we will be writing the configuration options which will be avaliable for our
+INISource. The INISourceConfig will be decorated by config. Here we have provided
+three configuration options.
+
+Create Source
+-------------
+
+**dffml_source_ini/misc.py**
+
+.. literalinclude:: ../../../dffml/source/ini.py
+    :lines: 18-24
+
+Here we have added the entrypoint to INISource class as "ini". The new Source should inherit
```

Comment:
We probably want to mention that there already is an `ini` source in dffml, so users will want to name theirs `myini` or something.

Reply:
sure